frontend/src/app/shared/models/IUser.ts

```diff
@@ -0,0 +1,19 @@
+import { LanguageLevel } from '@shared/models/enums/LanguageLevel';
+import { Sex } from '@shared/models/enums/Sex';
+
+export interface IUser {
+    uid?: string;
+    country?: number;
+    language?: number;
+    timezone?: number;
+    firstName: string;
+    lastName: string;
+    age?: number;
+    email: string;
+    imagePath?: string;
+    sex?: Sex;
+    languageLevel?: LanguageLevel;
+    status?: number;
+    isSubscribed?: boolean;
+    isBanned?: boolean;
```

Comment:
Instead of doing properties optional it would be better to use Partial in a place where you don't need all required fields